devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java

```diff
@@ -110,8 +110,11 @@ static void setProjectModels(QuarkusBootstrapMojo mojo, BootstrapMavenContextCon
      * @return raw POM
      */
     private static Model getRawModel(MavenProject mp) {
-        final Model model = mp.getOriginalModel();
+        Model model = mp.getOriginalModel();
```

Comment:
Why not clone the model immediately? Isn't here an unconditional modification in line 125?